ml-agents/mlagents/trainers/torch/encoders.py

```diff
@@ -9,9 +9,9 @@
 class Normalizer(nn.Module):
     def __init__(self, vec_obs_size: int):
         super().__init__()
-        self.normalization_steps = torch.tensor(1)
-        self.running_mean = torch.zeros(vec_obs_size)
-        self.running_variance = torch.ones(vec_obs_size)
+        self.register_buffer("normalization_steps", torch.tensor(1))
```

Comment:
I think we need to determine if we want to use `Parameter(require_grad=False)` or `register_buffer`. I tend to prefer the first one since it is more intuitive what it means. I think @dongruoping is also using `Parameter` in her PR.

Reply:
I do agree Parameter is more intuitive. Went with `register_buffer` b/c it's what's used by PyTorch in their own code. E.g. running mean and variance in the Batch Norm module: https://pytorch.org/docs/stable/_modules/torch/nn/modules/batchnorm.html#BatchNorm1d

I agree though we should make it the same across all our code. @dongruoping what do you think?

Reply:
They're pretty much the same effectively but I think Pytorch made this buffer thing to avoid the impression of this variable being something "trainable" and also avoid forgetting to set require_grad=False. The other benefit is that these will not need be passed to the optimizer and then be skipped by optimizer when updating. Not big issue.
The name `buffer` is not the best though I agree.
I'm good with both as long as we all use the same one.

Reply:
Actually if it's used by PyTorch maybe it would be good to follow.